src/rust/engine/graph/src/entry.rs

```diff
@@ -423,6 +423,25 @@ impl<N: Node> Entry<N> {
             None,
             previous_result,
           )
+        } else if !self.node.content().cacheable() {
+          // Notify waiters
+          for waiter in waiters {
+            let _ = waiter.send(
+              previous_result
+                .clone()
+                .expect("I expected a result")
+                .map(|res| (res, generation)),
+            );
+          }
+          // Change to not started, with:
+          //  a new run,
+          //  the same generation, because we ignore it?
```

Comment:
You're incrementing the generation here, which is correct: we should treat every run of the node as a new generation, and never store its result.